paper/src/scrapers/public_hamburg/hamburg_public_search_detail_scraper.rs

```diff
@@ -6,35 +6,24 @@ use crate::model::Location;
 use crate::model::SearchResultDetail;
 use crate::scrapers::text_provider::TextProvider;
 use scraper::Selector;
+use std::future::Future;
 
 #[derive(uniffi::Object)]
 pub(crate) struct HamburgPublicSearchDetailScraper {}
 
 impl HamburgPublicSearchDetailScraper {
-    pub(crate) async fn search_result_detail_from(
+    pub(crate) fn search_result_detail_from(
         document: scraper::Html,
-    ) -> Option<SearchResultDetail> {
+    ) -> impl Future<Output = SearchResultDetail> + Send {
```

Comment:
This confused be for a moment that the 2 scrapers _Opc4v2_13Vzg6SearchDetailScraper/HamburgPublicSearchDetailScraper_ have similar API with different return types. Nothing you need to change, just something I stumbled upon.